lib/codecs/src/decoding/format/syslog.rs

```diff
@@ -47,8 +50,8 @@ impl SyslogDeserializerConfig {
             .optional_field("msgid", Kind::bytes(), None)
             .optional_field("procid", Kind::integer().or_bytes(), None)
             // "structured data" in a syslog message can be stored in any field, but will always be
-            // a string.
-            .unknown_fields(Kind::bytes())
+            // a map of string -> string.
```

Comment:
Aren't some of the values not strings?

Reply:
I believe in the sdata section of syslog protocol they should all be strings.

Reply:
Doh - incorrectly linked this to the optional fields above 🙃

Reply:
👍 it's a bit confusing because the sdata fields are nested under the sdelement name and so we don't know the field name ahead of time which is why `unknown_fields` is used here.